zlib-rs/src/allocate.rs

```diff
@@ -340,10 +340,10 @@ impl Allocator<'_> {
         if !ptr.is_null() {
             // Special case for the Rust `alloc` backed allocator
             #[cfg(feature = "rust-allocator")]
-            if self.zfree == Allocator::RUST.zfree {
+            if self.zfree == self::RUST.zfree {
                 assert_ne!(len, 0, "invalid size for {ptr:?}");
                 let mut size = core::mem::size_of::<T>() * len;
-                return (Allocator::RUST.zfree)(&mut size as *mut usize as *mut c_void, ptr.cast());
+                return (self::RUST.zfree)(&mut size as *mut usize as *mut c_void, ptr.cast());
```

Comment:
```suggestion
                return (RUST.zfree)(&mut size as *mut usize as *mut c_void, ptr.cast());
```

and elsewhere in the `allocate` module.